crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -295,7 +295,7 @@ impl Felt {
 
     /// Quotient and remainder between `self` and `rhs`.
     pub fn div_rem(&self, rhs: &NonZeroFelt) -> (Self, Self) {
-        let (q, r) = self.0.representative().div_rem(&rhs.0.representative());
+        let (q, r) = self.representative().div_rem(&rhs.0.representative());
```

Comment:
doesn't this change the behaviour?

Reply:
the `self.representative()` function returns `self.0.representative()` so it's the same. 
I had to add the public function `representative` to get the representation of a `Felt` for the `operate_with_self` function in order to multiply the point with itself.
Adding `representative()` function to `Felt` and replace the old way of using it make the code cleaner.